test/test-multi-strategy-erc4626.js

```diff
@@ -0,0 +1,603 @@
+    ).to.be.revertedWithCustomError(strategies[5], "Fail");
+
+    await expect(vault.connect(lp2).addStrategy(strategies[5], encodeDummyStorage({})))
+      .to.emit(vault, "StrategyAdded")
+      .withArgs(strategies[5], 3);
+    expect(await vault.depositQueue()).to.deep.equal([2, 1, 3, 4].concat(Array(MAX_STRATEGIES - 4).fill(0)));
+    expect(await vault.withdrawQueue()).to.deep.equal([3, 1, 2, 4].concat(Array(MAX_STRATEGIES - 4).fill(0)));
+    await invariantChecks(vault);
+  });
+
+  it("It can add up to 32 strategies", async () => {
+    const { deployVault, lp2, DummyInvestStrategy, currency, admin, strategies } = await helpers.loadFixture(setUp);
+    const vault = await deployVault(30);
+    await invariantChecks(vault);
+    await vault.connect(admin).grantRole(getRole("STRATEGY_ADMIN_ROLE"), lp2);
+
+    // Add 31 works fine
+    await expect(vault.connect(lp2).addStrategy(strategies[30], encodeDummyStorage({})))
+      .to.emit(vault, "StrategyAdded")
+      .withArgs(strategies[30], 30);
+    await invariantChecks(vault);
+
+    // Add 32 works fine
+    await expect(vault.connect(lp2).addStrategy(strategies[31], encodeDummyStorage({})))
+      .to.emit(vault, "StrategyAdded")
+      .withArgs(strategies[31], 31);
+    await invariantChecks(vault);
+
+    const strategy33 = await DummyInvestStrategy.deploy(currency);
+
+    // Another one fails
+    await expect(vault.connect(lp2).addStrategy(strategy33, encodeDummyStorage({}))).to.be.revertedWithCustomError(
+      vault,
+      "InvalidStrategiesLength"
+    );
+    await invariantChecks(vault);
+  });
+
+  it("It can removeStrategy only if doesn't have funds", async () => {
+    const { deployVault, lp, lp2, currency, admin, strategies } = await helpers.loadFixture(setUp);
+    const vault = await deployVault(3, undefined, [1, 0, 2], [2, 0, 1]);
+    await currency.connect(lp).approve(vault, MaxUint256);
+    await vault.connect(admin).grantRole(getRole("LP_ROLE"), lp);
+    await expect(vault.connect(lp).mint(_A(100), lp)).not.to.be.reverted;
+    await invariantChecks(vault);
+
+    expect(await vault.totalAssets()).to.be.equal(_A(100));
+    // Check money went to strategy[3]
+    expect(await currency.balanceOf(await strategies[1].other())).to.be.equal(_A(100));
+
+    await expect(vault.connect(lp2).removeStrategy(0, false)).to.be.revertedWith(
+      accessControlMessage(lp2, null, "STRATEGY_ADMIN_ROLE")
+    );
+
+    await vault.connect(admin).grantRole(getRole("STRATEGY_ADMIN_ROLE"), lp2);
+
+    await expect(vault.connect(lp2).removeStrategy(33, false)).to.be.revertedWithCustomError(vault, "InvalidStrategy");
+    await expect(vault.connect(lp2).removeStrategy(5, false)).to.be.revertedWithCustomError(vault, "InvalidStrategy");
+    await expect(vault.connect(lp2).removeStrategy(1, false)).to.be.revertedWithCustomError(
+      vault,
+      "CannotRemoveStrategyWithAssets"
+    );
+
+    await expect(vault.connect(lp2).removeStrategy(0, false))
```

Comment:
If you make the happy path removal with strategy 1 instead of 0 it would improve coverage, because this is not testing the case where withdraw queue elements **don't** need to be reindexed. i.e., when this conditions isn't entered:

```js
        if (_withdrawQueue[i] > (strategyIndex + 1)) {
          // If index is greater, substract one
          _withdrawQueue[i] -= 1;
        }
```